cats/__init__.py

```diff
@@ -27,79 +56,103 @@ def main(arguments=None):
-#            "-m",
-#            f"{runtime:%m%d%H%M}",
-#        ]
-#    )
+    sys.stderr.write("\n" + str(best_window) + "\n")
 
     sys.stderr.write(f"Best job start time: {best_window.start}\n")
     print(f"{best_window.start:%Y%m%d%H%M}")  # for POSIX compatibility with at -t
 
+    ################################
+    ## Calculate carbon footprint ##
+    ################################
+
     if args.jobinfo:
-        jobinfo = validate_jobinfo(args.jobinfo)
-        if not jobinfo:
-            print("ERROR: job info parsing failed, exiting now")
-            exit(1)
-        if not config:
-            print("ERROR: config file not found, exiting now")
-            exit(1)
-        now_avg_ci = avg_carbon_intensity(
-            data=CI_forecast, start=datetime.now(), runtime=timedelta(args.duration)
-        )
-        estim = greenAlgorithmsCalculator(
-            config=config,
-            runtime=timedelta(minutes=args.duration),
-            averageBest_carbonIntensity=best_window.value, # TODO replace with real carbon intensity
-            averageNow_carbonIntensity=now_avg_ci,
-            **jobinfo,
-        ).get_footprint()
-
-        print(" -!-!- Carbon footprint estimation is a work in progress, coming soon!")
-        # Commenting these out while waiting for real carbon intensities
-        # print(f"Estimated emmissions for running job now: {estim.now}")
-        # msg = (
-        #     f"Estimated emmissions for running delayed job: {estim.best})"
-        #     f" (- {estim.savings})"
-        # )
-        # print(msg)
+        jobinfo = validate_jobinfo(args.jobinfo, expected_partition_names=config['partitions'].keys())
+
+        if not (jobinfo and config):
+            sys.stderr.write("Not enough information to estimate total carbon footprint, both --jobinfo and config files are needed.\n")
+        else:
+            now_avg_ci = avg_carbon_intensity(
+                data=CI_forecast, start=datetime.now(), runtime=timedelta(args.duration)
+            )
+            estim = greenAlgorithmsCalculator(
+                config=config,
+                runtime=timedelta(minutes=args.duration),
+                averageBest_carbonIntensity=best_window.value, # TODO replace with real carbon intensity
+                averageNow_carbonIntensity=now_avg_ci,
+                **jobinfo,
+            ).get_footprint()
+
+            sys.stderr.write("\n -!-!- Carbon footprint estimation is a work in progress, coming soon!\n")
+            # Commenting these out while waiting for real carbon intensities
+            # print(f"Estimated emmissions for running job now: {estim.now}")
+            # msg = (
+            #     f"Estimated emmissions for running delayed job: {estim.best})"
+            #     f" (- {estim.savings})"
+            # )
+            # print(msg)
+    else:
+        sys.stderr.write("Not enough information to estimate total carbon footprint, both --jobinfo and config files are needed.\n")
```

Comment:
It would be good to define this message earlier under a variable for re-use, since we define the same string twice to use for `stderr` in this block and that way we only need to tweak it once if we want to change it (etc.).

Reply:
Yep good call, done